packages/nextjs/utils/service-workers/index.ts

```diff
@@ -9,41 +12,57 @@ export const unregisterServiceWorkers = async () => {
   await Promise.all(registrations.map(r => r.unregister()));
 };
 
-export const subscribe = async () => {
+export const askForPermission = async () => {
   try {
-    const swRegistration = await navigator.serviceWorker.getRegistration("/sw.js");
     const premissionResult = await window?.Notification.requestPermission();
     if (premissionResult === "denied") alert("Permission already denied, please enable notifications manually");
+  } catch (err) {
+    console.error("Inside askForPermission function: ", err);
+    throw new Error(`Error while allowing for notifications`);
+  }
+};
+
+export const subscribeToNotifications = async () => {
+  let notificationId = null;
+  try {
+    notificationId = notification.loading("Waiting for serviceWorker");
+    await navigator.serviceWorker.ready;
+    notification.remove(notificationId);
```

Comment:
Notice you won't get "Waiting for serviceWorker" toast when you are in desktop or even good mobile chrome browser 

but this shows only in my testing in Safari browser of Iphone (since you can PWA on thorough Safari in IOS I think) and safari kind of takes time to register SW when you add it very first time after that its really smooth

We could update "Waiting for serviceWorker" with a kindof non-techfriendly message  😅🙌